digitalearthau/move.py

```diff
@@ -200,39 +213,55 @@ def _do_copy(self, dry_run=True, checksum=True):
         dataset_path = self.source_path
 
         if checksum:
-            successful_checksum = _verify_checksum(self.log, self.source_metadata_path,
+            successful_checksum = _verify_checksum(self.log, self.from_metadata_path,
                                                    dry_run=dry_run)
             self.log.info("checksum.complete", passes_checksum=successful_checksum)
             if not successful_checksum:
-                raise RuntimeError("Checksum failure on " + str(self.source_metadata_path))
+                raise RuntimeError("Checksum failure on " + str(self.from_metadata_path))
 
         if dataset_path.is_dir():
-            log.debug("copy.mkdir", dest=dest_path.parent)
-            fileutils.mkdir_p(str(dest_path.parent))
-
-            # We don't want to risk partially-copied packaged left on disk, so we copy to a tmp dir in same
-            # folder and then atomically rename into place.
-            tmp_dir = Path(tempfile.mkdtemp(prefix='.dea-mv-', dir=str(dest_path.parent)))
-            try:
-                tmp_package = tmp_dir.joinpath(dataset_path.name)
-                log.info("copy.put", src=dataset_path, tmp_dest=tmp_package)
-                if not dry_run:
-                    shutil.copytree(str(dataset_path), str(tmp_package))
-                    log.debug("copy.put.done")
-                    os.rename(str(tmp_package), str(dest_path))
-                    log.debug("copy.rename.done")
-
-                    # It should have been contained within the dataset, see the check in the constructor.
-                    assert self.dest_metadata_path.exists()
-            finally:
-                log.debug("tmp_dir.rm", tmp_dir=tmp_dir)
-                shutil.rmtree(str(tmp_dir), ignore_errors=True)
+            self.copy_directory(dataset_path, dest_path, dry_run, log)
+        elif self.dest_path == self.dest_metadata_path:  # Metadata is contained within the dataset file. eg. *.nc
+            self.copy_file(dataset_path, dest_path, log)
         else:
-            # .nc files and sibling files
+            # Datasets that are dataset file + sibling or metadata separate to data
             raise NotImplementedError("TODO: dataset files not yet supported")
 
         return self.dest_uri
 
+    def copy_file(self, from_, to, log):
+        to_directory = to.parent
+        log.debug("copy.mkdir", dest=to_directory)
+        fileutils.mkdir_p(to.parent)
+        # We don't want to risk partially-copied files left on disk, so we copy to a tmp name
+        # then atomically rename into place.
+        tmp_name = tempfile.mktemp(prefix='.dea-mv-', dir=to_directory)
+        log.info("copy.put", src=from_, tmp_dest=tmp_name)
+        shutil.copy(from_, tmp_name)
+        log.debug("copy.put.done")
+        os.rename(tmp_name, to)
```

Comment:
It's worth adding a `finally` block, similar to `copy_directory`, so that we don't leave the tmp file on disk if the copy fails mid-way-through.